src/api/app/controllers/webui/users_controller.rb

```diff
@@ -11,13 +11,37 @@ def index
   end
 
   def show
-    @iprojects = @displayed_user.involved_projects.pluck(:name, :title)
-    @ipackages = @displayed_user.involved_packages.joins(:project).pluck(:name, 'projects.name as pname')
-    @owned = @displayed_user.owned_packages
     @groups = @displayed_user.groups
     @role_titles = @displayed_user.roles.global.pluck(:title)
     @account_edit_link = CONFIG['proxy_auth_account_page']
 
+    if Flipper.enabled?(:user_profile_redesign, User.possibly_nobody)
+      filters = params.slice(:search_text, :involved_projects, :involved_packages,
+                             :role_owner, :role_maintainer, :role_bugowner, :role_reviewer, :role_downloader, :role_reader)
+
+      filters[:search_text] = filters[:search_text]&.strip
```

Comment:
Please extract this logic to a testable object, either a method or a service object.

Reply:
Codeclimate was complaining. I was working on it. I moved this logic to `adjust_filters`. Please take a look again.